tests/integration/test_charm_no_runner.py

```diff
@@ -1,246 +0,0 @@
-    assert await get_repo_policy_compliance_pip_info(unit) is not None
-
-
-@pytest.mark.asyncio
-@pytest.mark.abort_on_fail
-async def test_update_dependencies_action_old_service(
-    model: Model, app_no_runner: Application
-) -> None:
-    """
-    arrange: Replace repo-policy-compliance service installation to a older version.
-    act: Run update-dependencies action.
-    assert:
-        a. Service is installed in the charm.
-        b. Action flushed the runners.
-    """
-    unit = app_no_runner.units[0]
-    latest_version_info = await get_repo_policy_compliance_pip_info(unit)
-
-    await install_repo_policy_compliance_from_git_source(
-        unit, REPO_POLICY_COMPLIANCE_VER_0_2_GIT_SOURCE
-    )
-    assert await get_repo_policy_compliance_pip_info(unit) != latest_version_info
-
-    action = await unit.run_action("update-dependencies")
-    await action.wait()
-    await model.wait_for_idle(status=ACTIVE)
-
-    assert action.results["flush"] == "True"
-    assert await get_repo_policy_compliance_pip_info(unit) is not None
-
-
-@pytest.mark.asyncio
-@pytest.mark.abort_on_fail
-async def test_update_dependencies_action_on_runner_binary(
-    model: Model, app_no_runner: Application
-) -> None:
-    """
-    arrange: Remove runner binary if exists.
-    act:
-        1. Run update-dependencies action.
-        2. Run update-dependencies action.
-    assert:
-        1.  a. Runner binary exists in the charm.
-            b. Action flushed the runners.
-        2.  a. Runner binary exists in the charm.
-            b. Action did not flushed the runners.
-    """
-    unit = app_no_runner.units[0]
-
-    await remove_runner_bin(unit)
-
-    action = await unit.run_action("update-dependencies")
-    await action.wait()
-    await model.wait_for_idle(status=ACTIVE)
-
-    # The runners should be flushed on update of runner binary.
-    assert action.results["flush"] == "True"
-
-    assert await check_runner_binary_exists(unit)
-
-    action = await unit.run_action("update-dependencies")
-    await action.wait()
-    await model.wait_for_idle(status=ACTIVE)
-
-    # The runners should be flushed on update of runner binary.
-    assert action.results["flush"] == "False"
-
-    assert await check_runner_binary_exists(unit)
-
-
-@pytest.mark.asyncio
-@pytest.mark.abort_on_fail
-async def test_check_runners_no_runners(app_no_runner: Application) -> None:
-    """
-    arrange: A working application with no runners.
-    act: Run check-runners action.
-    assert: Action returns result with no runner.
-    """
-    unit = app_no_runner.units[0]
-
-    action = await unit.run_action("check-runners")
-    await action.wait()
-
-    assert action.results["online"] == "0"
-    assert action.results["offline"] == "0"
-    assert action.results["unknown"] == "0"
-    assert not action.results["runners"]
-
-
-@pytest.mark.asyncio
-@pytest.mark.abort_on_fail
-async def test_reconcile_runners(model: Model, app_no_runner: Application) -> None:
```

Comment:
These two test could be applicable for Openstack (but were not included in the integration test workflow).